pkg/cloudprovider/aws/fleet/capacity.go

```diff
@@ -76,8 +76,8 @@ func (c *Capacity) Create(ctx context.Context, constraints *cloudprovider.Capaci
 			}},
 		}},
 	})
-	if err != nil {
-		return nil, fmt.Errorf("creating fleet, %w", err)
+	if err != nil || len(createFleetOutput.Errors) > 0 {
```

Comment:
Wow this is such a funky API. Nice catch. I think this error message may end up being confusing. Perhaps we can split into two if statements?
```
if err != nil {
 return ...  "creating fleet %w"
}
if len(createFleetOutput.Errors) {
  return ... "with create fleet errors %w" 
}
```